packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2289,7 +2289,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     parseClassProperty(node: N.ClassProperty): N.ClassProperty {
       this.parseClassPropertyAnnotation(node);
 
-      if (node.declare && this.match(tt.eq)) {
+      if ((node.declare || this.state.isDeclareContext) && this.match(tt.eq)) {
```

Comment:
Does `node.declare` imply `this.state.isDeclareContext`? If so we can remove `node.declare` checks.

Reply:
No, `this.state.isDeclareContext` is not for class member with `declare` modifier.